src/components/MobileMenu.tsx

```diff
@@ -0,0 +1,93 @@
+import { observer } from 'mobx-react-lite';
+import Link from 'next/link';
+import { edgeConfigStore } from '../stores/edge-config-store';
+
+interface MobileMenuProps {
+  isOpen: boolean;
+  close: () => void;
+}
+
+const MobileMenu: React.FC<MobileMenuProps> = ({ isOpen, close }) => {
+  const { configData } = edgeConfigStore;
+
+  if (!configData) {
+    return <div>Loading...</div>;
+  }
+
+  const categories = configData.categories;
+
+  const uniqueBrands = new Set<string>();
+  Object.keys(categories).forEach((categoryKey) => {
+    const category = categories[categoryKey];
+    category.brands.forEach((brand) => {
+      uniqueBrands.add(brand);
+    });
+  });
+
+  const handleCloseMenu = (e: React.MouseEvent) => {
+    e.stopPropagation();
+  };
+
+  return (
+    <div
+      className={`md:hidden bg-gray-800 text-white w-80 fixed top-0 right-0 h-screen z-50 transform transition-transform ${
+        isOpen ? 'translate-x-0' : 'translate-x-full'
+      }`}
+      onClick={close}
+    >
+      <div
+        className="h-full overflow-y-auto ps-4 bg-gray-800 custom-scrollbar"
+        onClick={handleCloseMenu}
+      >
+        <button
+          onClick={close}
+          className="text-white text-2xl fixed top-7.5 right-4 z-50"
+        >
+          <i className="material-icons">close</i>
+        </button>
+
+        <div className="mt-12">
+          {['men', 'women', 'kids'].map((categoryKey) => {
+            const category = categories[categoryKey];
+
+            if (!category) return null;
+
+            return (
```

Comment:
generally we'd do something like 
```suggestion
            if (!category) return null;

            return  !category ? null : (
```

Except with `category` instead of `!category` I didn't want to contextualize the entire return statement here.

Reply:
so wait, I'm confused. I think just how the comment code is left. you're suggesting it should be 

```suggestion
             return !category ? null : (
```
          
removing `if (!category) return null;` entirely and just putting it in that one line and then go into the mapping, correct?

Reply:
Actually, something like

`category && <Component />`